src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedContainer.cs

```diff
@@ -123,16 +125,40 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
  
         public override ImmutableArray<Symbol> GetMembers()
         {
+            var builder = ArrayBuilder<Symbol>.GetInstance();
+            if (this.TypeKind == TypeKind.Struct)
+            {
+                // Add an extra object field for EnC.
+                builder.Add(GetStructFieldCaptureObject());
+            }
+
             Symbol constructor = this.Constructor;
-            return (object)constructor == null ? ImmutableArray<Symbol>.Empty : ImmutableArray.Create(constructor);
+            if ((object)constructor != null)
+            {
+                builder.Add(constructor);
+            }
+
+            return builder.ToImmutableAndFree();
         }
 
         public override ImmutableArray<Symbol> GetMembers(string name)
         {
+            if (this.TypeKind == TypeKind.Struct && name == StructFieldCaptureName)
+            {
+                // Add an extra object field for EnC.
+                return ImmutableArray.Create(GetStructFieldCaptureObject());
+            }
+
             var ctor = Constructor;
             return ((object)ctor != null && name == ctor.Name) ? ImmutableArray.Create<Symbol>(ctor) : ImmutableArray<Symbol>.Empty;
         }
 
+        private Symbol GetStructFieldCaptureObject()
+        {
+            TypeSymbol objectType = this.DeclaringCompilation.GetSpecialType(SpecialType.System_Object);
+            return new SynthesizedFieldSymbol(this, objectType, StructFieldCaptureName, isPublic: false, isReadOnly: false, isStatic: false);
```

Comment:
This method needs to be idempotent. It should never be possible to get two fields out of this method that are not reference equal to each other.